chart/templates/_helpers.yaml

```
@@ -23,7 +23,40 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 If release name contains chart name it will be used as a full name.
 */}}
 {{- define "airflow.fullname" -}}
-  {{- if .Values.fullnameOverride }}
+  {{- if not .Values.useStandardNaming }}
+    {{- .Release.Name }}
+  {{- else if .Values.fullnameOverride }}
+    {{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+  {{- else }}
+    {{- $name := default .Chart.Name .Values.nameOverride }}
+    {{- if contains $name .Release.Name }}
+      {{- .Release.Name | trunc 63 | trimSuffix "-" }}
+    {{- else }}
+      {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name for objects that already contained airflow in the name
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "airflow.name" -}}
+  {{ if .Values.fullnameOverride }}
+    {{- printf "%s" .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+  {{- else }}
+    {{- $name := default .Chart.Name .Values.nameOverride }}
+    {{- if contains $name .Release.Name }}
+      {{- .Release.Name | trunc 63 | trimSuffix "-" }}
+    {{- else }}
+      {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
```

```suggestion
      {{- printf "%s-airflow-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
```

Did you intend to do this so the `airflow` part remains? Otherwise, I don't think we need to keep this. I think all of the old objects with "airflow" in the name can be renamed safely.

---

if we add -airflow-" we endup with double airflow e.g. `release-name-airflow-airflow-metadata` as Chart.Name is already airflow - that's why I didn't kept airflow there

---

I guess I misunderstood the intent of this helper. I thought it existed to keep airflow in the existing object names (which, as written, we aren't, per the tests).

If we consider the non-standard naming case, we are already renaming the objects, so why don't we just start using `airflow.fullname`? Do we need to have this special helper?

---

yeah I think we can use fullname for all instead. I've removed airflow.name and kept consistency in https://github.com/apache/airflow/pull/31066/commits/b790f98d5a2634fa30e115d2ceb1598869cebcf9 😃